src/main/java/uk/gov/hmcts/ccd/domain/model/search/elasticsearch/ElasticsearchRequest.java

```diff
@@ -59,22 +93,64 @@ public JsonNode getSource() {
                 .map(JsonNode::asText)
                 .map(this::getFieldId)
                 .filter(Objects::nonNull)
-                .collect(Collectors.toList());
+                .collect(toList());
         }
 
         return Collections.emptyList();
     }
 
     public boolean hasQuery() {
-        return searchRequest.has(QUERY);
+        return nativeSearchRequest.has(QUERY);
     }
 
     public JsonNode getQuery() {
-        return searchRequest.get(QUERY);
+        return nativeSearchRequest.get(QUERY);
+    }
+
+    public boolean hasSupplementaryData() {
+        return supplementaryData != null && supplementaryData.isArray();
+    }
+
+    public JsonNode getSupplementaryData() {
+        return supplementaryData;
+    }
+
+    public void setSupplementaryData(JsonNode node) {
+        if (node.isArray() && arrayContainsOnlyText((ArrayNode) node)) {
+            this.supplementaryData = (ArrayNode) node;
+        } else {
+            throw new BadSearchRequest("Provided supplementary_data must be an array of text fields.");
+        }
     }
 
+    /**
+     * Creates a JSON string representing the Elasticsearch request object.
+     * Custom properties supported by CCD will be merged appropriately to generate a native Elasticsearch request.
+     * @return JSON string representing the Elasticsearch request object.
+     */
     public String toJson() {
```

Comment:
this is now doing more than simply converting to Json. Shall we rename to:
`toFinalRequest()` or sim?